docs/reference/sql/language/data-types.asciidoc

```diff
@@ -16,21 +16,21 @@ s|SQL precision
 
 3+h| Core types
 
-| <<null-value, `null`>>      | null        | 0
-| <<boolean, `boolean`>>      | boolean     | 1
-| <<number, `byte`>>          | tinyint     | 3
-| <<number, `short`>>         | smallint    | 5
-| <<number, `integer`>>       | integer     | 10
-| <<number, `long`>>          | bigint      | 19
-| <<number, `double`>>        | double      | 15
-| <<number, `float`>>         | real        | 7
-| <<number, `half_float`>>    | float       | 16
-| <<number, `scaled_float`>>  | float       | 19
-| <<keyword, `keyword`>>      | varchar     | based on <<ignore-above>>
-| <<text, `text`>>            | varchar     | 2,147,483,647
-| <<binary, `binary`>>        | varbinary   | 2,147,483,647
-| <<date, `date`>>            | timestamp   | 24
-| <<ip, `ip`>>                | varchar     | 39
+| <<null-value, `null`>>      | null                  | 0
+| <<boolean, `boolean`>>      | boolean               | 1
+| <<number, `byte`>>          | tinyint               | 3
+| <<number, `short`>>         | smallint              | 5
+| <<number, `integer`>>       | integer               | 10
+| <<number, `long`>>          | bigint                | 19
+| <<number, `double`>>        | double                | 15
+| <<number, `float`>>         | real                  | 7
+| <<number, `half_float`>>    | float                 | 16
+| <<number, `scaled_float`>>  | float                 | 19
+| <<keyword, `keyword`>>      | varchar               | based on <<ignore-above>>
+| <<text, `text`>>            | varchar               | 2,147,483,647
+| <<binary, `binary`>>        | varbinary             | 2,147,483,647
+| <<date, `date`>>            | timestamp/datetime    | 24
```

Comment:
Maybe it makes sense to add another column so the table contains ES/ES-SQL/SQL types. It might be redundant but it might make the `datetime` case clearer - can be done in a separate PR.

Reply:
Opened issue to track it: #37519